addons/mrp/models/stock_move.py

```diff
@@ -707,3 +698,36 @@ def _get_relevant_state_among_moves(self):
                         for move in self):
             res = 'assigned'
         return res
+
+    def _description_bom_line(self, attribute='', model=''):
+        bom_line_description = {}
+        moves_to_update = self
+        moves_to_return = self.env['stock.move']
+
+        def get_description(moves):
+            if not moves:
+                return
+            for bom in moves.bom_line_id.bom_id:
+                if bom.type != 'phantom':
+                    continue
+                line_ids = moves.bom_line_id.filtered(lambda line: line.bom_id == bom).mapped('id')
+                total = len(line_ids)
+                name = bom.display_name
+                for i, line_id in enumerate(line_ids):
+                    bom_line_description[line_id] = '%s - %d/%d' % (name, i + 1, total)
+
+        if not attribute or not model:
+            get_description(self)
+        else:
+            move_from_model = defaultdict(lambda: self.env['stock.move'])
+            for move in self:
+                move_from_model[move[attribute]] += move
+            for attribute_id, moves in move_from_model.items():
+                get_description(moves)
+            moves_to_return = move_from_model[self.env[model]]
```

Comment:
I think it's not used. You could keep it simple as before but with the fix on length base on move and not bom lines